src/rdkafka_msgset_reader.c

```diff
@@ -692,8 +701,129 @@ rd_kafka_msgset_reader_msg_v2 (rd_kafka_msgset_reader_t *msetr) {
                 return RD_KAFKA_RESP_ERR_NO_ERROR; /* Continue with next msg */
         }
 
-        rd_kafka_buf_read_bytes_varint(rkbuf, &hdr.Key);
+        /* Handle control messages */
+        if (msetr->msetr_v2_hdr->Attributes & RD_KAFKA_MSGSET_V2_ATTR_CONTROL) {
+                struct {
+                        int64_t KeySize;
+                        int16_t Version;
+                        int16_t Type;
+                } ctrl_data;
+                rd_kafka_aborted_txn_start_offsets_t node, *node_ptr;
```

Comment:
`node` is the name of the avl element, which is an implementation detail, this variable should be named something pertaining to txn_start_offsets_t  (same comment for broker.c)

Reply:
i've changed this (and in broker.c), though I feel it is harder to follow now with the extra characters and also that the word node is more informative/useful, because it matches a key aspect of the algorithm, which the reader will be trying to follow when reading this. now this information is obfuscated in favor of something that provides less value.

Reply:
sorry I've changed this back to node again because i think it's making things worse (away from txn_start_offsets / _ptr). If you want this changed, tell me precisely what name you want and I'll change it.